src/Microsoft.OData.Core/UriParser/Parsers/ODataPathParser.cs

```diff
@@ -1086,13 +1191,14 @@ private void CreateNextSegment(string text)
              * Treat as a key.
              */
 
+
+            ODataPathSegment previous = this.parsedSegments[this.parsedSegments.Count - 1];
             // $value

```

Comment:
Why move this?

Reply:
You mean why move the TryCreateEscapeFunctionSegment? I did that to not ExtractIdentifierAndParenthesis proactively for escape function segment. If it is successful, we may not need to at this point. 

---
In reply to: [369818816](https://github.com/OData/odata.net/pull/1621#discussion_r369818816) [](ancestors = 369818816)

Reply:
See iteration 22: you had moved ODataPathSegment previous = this.parsedSegments[this.parsedSegments.Count - 1]; from line 1202 (immediately before using it to check previous.TargetKind == RequestTargetKind.Primitive, where is is first used) to line 1195 (before TryCreateValueSegment). It looks like it moved back in iteration 23.

---
In reply to: [369827716](https://github.com/OData/odata.net/pull/1621#discussion_r369827716) [](ancestors = 369827716,369818816)